pylint/checkers/base_checker.py

```diff
@@ -43,15 +45,84 @@ def __init__(self, linter=None):
         OptionsProviderMixIn.__init__(self)
         self.linter = linter
 
+    def __gt__(self, other):
+        """Permit to sort a list of Checker by name."""
+        return "{}{}".format(self.name, self.msgs).__gt__(
+            "{}{}".format(other.name, other.msgs)
+        )
+
+    def __repr__(self):
+        status = "Checker" if self.enabled else "Disabled checker"
+        msgids = [id for id in self.msgs]
```

Comment:
Let's use a different variable name, as `id` is shadowing the builtin `id()` callable. Alternatively this can just be `list(msgs)`

Reply:
Good catch, I replaced by `self.msgs.keys()`.